pkg/services/searchV2/index.go

```diff
@@ -846,7 +846,14 @@ func (l sqlDashboardLoader) loadAllDashboards(ctx context.Context, limit int, or
 				attribute.Int64("lastID", lastID),
 			))
 
-			rows := make([]*dashboardQueryResult, 0)
+			var rows []*dashboardQueryResult
+			if dashboardUID != "" { // only expecting a single result
+				// xorm will append if there is more than one
+				rows = make([]*dashboardQueryResult, 0, 1)
+			} else {
+				// start with an arbitrary but reasonable number of dashboards
+				rows = make([]*dashboardQueryResult, 0, 1000)
```

Comment:
Instead of arbitrary, should we use the `limit` that is passed in?

Reply:
Yes, that's a good point, I can use that when limit isn't set to 0, but we still need a good default value when we're returning all.

Reply:
Oh, even better, now I see the default `limit` is tied to `DashboardLoadingBatchSize`, nice! Already configurable :)